lake/index/rule.go

```diff
@@ -176,3 +176,35 @@ func (a *AggRule) RuleKeys() field.List {
 	// XXX can get these by analyzing the compiled script
 	return nil
 }
+
+// NewLookupKey creates a Zed Record that can be used as a lookup key for an
+// index created for the provided Rule. The Values provided must be in order
+// with the Key in the rule that it will be paired with it.
+func NewLookupKey(zctx *zed.Context, r Rule, values []zed.Value) (*zed.Record, error) {
```

Comment:
Is this going to be used in other packages? If not, don't export.

Reply:
good point